aws-lc-rs/benches/kem_benchmark.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+use aws_lc_rs::{test, test_file};
+use criterion::{criterion_group, criterion_main, Criterion};
+
+#[allow(non_camel_case_types)]
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+enum Algorithm {
+    KYBER512_R3,
+}
+
+impl From<&str> for Algorithm {
+    fn from(value: &str) -> Self {
+        match value {
+            "KYBER512_R3" => Algorithm::KYBER512_R3,
+            _ => panic!("Unrecognized algorithm: '{value}'"),
+        }
+    }
+}
+
+pub struct KemConfig {
+    algorithm: Algorithm,
+    public_key: Box<[u8]>,
+    secret_key: Box<[u8]>,
+    ciphertext: Box<[u8]>,
+}
+
+impl KemConfig {
+    fn new(algorithm: &str, public_key: &[u8], secret_key: &[u8], ciphertext: &[u8]) -> Self {
+        KemConfig {
+            algorithm: Algorithm::from(algorithm),
+            public_key: Vec::from(public_key).into(),
+            secret_key: Vec::from(secret_key).into(),
+            ciphertext: Vec::from(ciphertext).into(),
+        }
+    }
+}
+
+macro_rules! benchmark_kem {
+    ( $pkg:ident ) => {
+        paste::item! {
+            mod [<$pkg _benchmarks>]  {
+
+                use $pkg::kem;
```

Comment:
Is the macro needed?  We use macros in our other benchmarks to help obtain a fair comparison of aws-lc-rs to ring (or openssl), the macro can ensure we perform the operations in the same/comparable way in both libraries.